drivers/i3c/mux/i3c_renesas_imx3112.c

```diff
@@ -0,0 +1,603 @@
+		if (err) {
+			return err;
+		}
+
+		/* Select new channel mask */
+		err = i3c_renesas_imx3112_write_reg(
+			mux_dev, offsetof(struct i3c_renesas_imx3112_registers, mux_select),
+			select_mask_reg);
+		if (err) {
+			return err;
+		}
+
+		/* Re-enable mux */
+		err = i3c_renesas_imx3112_write_reg(
+			mux_dev, offsetof(struct i3c_renesas_imx3112_registers, mux_config),
+			select_mask_reg);
+		if (err) {
+			return err;
+		}
+
+		/* Update runtime data */
+		data->mux_state.enabled = true;
+		data->mux_state.channel_mask = select_mask;
+	}
+	return 0;
+}
+
+static int i3c_renesas_imx3112_i2c_transfer(const struct device *channel_dev, struct i2c_msg *msgs,
+					    uint8_t num_msgs, uint16_t addr)
+{
+
+	struct i3c_renesas_imx3112_mux_data *data = get_mux_data_from_channel(channel_dev);
+	const struct i3c_renesas_imx3112_mux_config *config =
+		get_mux_config_from_channel(channel_dev);
+	const struct i3c_renesas_imx3112_channel_config *channel_config = channel_dev->config;
+	const struct device *bus_dev = config->bus;
+	int err;
+
+	err = k_mutex_lock(&data->lock, CHAN_LOCK_TIMEOUT);
+	if (err) {
+		return err;
+	}
+
+	err = i3c_renesas_imx3112_set_channel(channel_config->mux, channel_config->channel_mask);
+	if (err) {
+		goto end_trans;
+	}
+
+	err = i2c_transfer(bus_dev, msgs, num_msgs, addr);
+
+end_trans:
+	k_mutex_unlock(&data->lock);
+	if (err) {
+		LOG_ERR("Mux dev %s transfer failed with error %d", channel_dev->name, err);
+	}
+	return err;
+}
+
+static int i3c_renesas_imx3112_transfer(const struct device *channel_dev,
+					struct i3c_device_desc *target, struct i3c_msg *msgs,
+					uint8_t num_msgs)
+{
+	struct i3c_renesas_imx3112_mux_data *data = get_mux_data_from_channel(channel_dev);
+	const struct i3c_renesas_imx3112_mux_config *config =
+		get_mux_config_from_channel(channel_dev);
+	const struct i3c_renesas_imx3112_channel_config *channel_config = channel_dev->config;
+	const struct device *bus_dev = config->bus;
+	struct i3c_device_id pid = {.pid = target->pid};
+	int err;
+
+	err = k_mutex_lock(&data->lock, CHAN_LOCK_TIMEOUT);
+	if (err) {
+		return err;
+	}
+
+	err = i3c_renesas_imx3112_set_channel(channel_config->mux, channel_config->channel_mask);
+	if (err) {
+		goto end_trans;
+	}
+	/* Change the device bus from the mux to the controller to redirect the transaction */
+	target = i3c_device_find(bus_dev, &pid);
+	if (target == NULL) {
+		LOG_ERR("Mux dev %s failed to find target: 0x%llX", channel_dev->name,
+			(uint64_t)pid.pid);
+		return -ENXIO;
+	}
+	err = i3c_transfer(target, msgs, num_msgs);
+
+end_trans:
+	k_mutex_unlock(&data->lock);
+	if (err) {
+		LOG_ERR("Mux dev %s transfer failed with error %d", channel_dev->name, err);
+	}
+	return err;
+}
+
+static int i3c_renesas_imx3112_configure(const struct device *dev, enum i3c_config_type type,
+					 void *config)
+{
+	return -ENOSYS;
+}
+
+static int i3c_renesas_imx3112_i2c_configure(const struct device *dev, uint32_t dev_config)
+{
+	return -ENOSYS;
+}
```

Comment:
I don't think thesefunctions are even needed to be defined here at all, as if they are NULL in the api struct, the api will just return -ENOSYS for you anyways